test/api/sse-dashboard.test.js

```diff
@@ -0,0 +1,313 @@
+jest.mock("../../server_src/config/auth");
+
+const EventSource = require("eventsource");
+const { parse } = require("flatted/cjs");
+const dbHandler = require("../db-handler");
+const supertest = require("supertest");
+const getEndpoints = require("express-list-endpoints");
+const { setupTestApp } = require("../../app-test");
+
+let request;
+const routeBase = "/dashboardInfo/";
+const ssePath = "get/";
+
+beforeAll(async () => {
+  await dbHandler.connect();
+  const server = await setupTestApp();
+
+  const endpoints = getEndpoints(server);
+  expect(endpoints).toContainEqual({
+    methods: ["GET"],
+    middleware: ["anonymous", "anonymous"],
+    path: routeBase + ssePath
+  });
+  request = supertest(server);
+});
+
+describe("SSE-dashboard", () => {
+  it("should be able to be called with an EventSource", async (done) => {
+    const getRequest = request.get(routeBase + ssePath);
+    const url = getRequest.url;
+    expect(url).toBeTruthy();
+
+    const es = new EventSource(url);
+
+    // events.emit('test', 'test message')
+    es.onmessage = (e) => {
+      const parsedMsg = parse(e.data);
+
+      expect(parsedMsg).toMatchObject({
+        printerInformation: expect.any(Array), // TODO printerInformation instead of printersInformation?
+        currentOperations: {
+          operations: expect.any(Array), // TODO printerInformation instead of printersInformation?
+          count: {
+            printerCount: expect.any(Number),
+            complete: expect.any(Number),
+            offline: expect.any(Number),
+            active: expect.any(Number),
+            idle: expect.any(Number),
+            disconnected: expect.any(Number),
+            farmProgress: expect.any(Number),
+            farmProgressColour: expect.any(String)
+          }
+        },
+        dashStatistics: {
+          currentUtilisation: expect.any(Array),
+          currentStatus: expect.any(Array), // TODO 3x null?,
```

Comment:
Please check why this is so and if its a problem or not

Reply:
Those been an array do you mean?

Reply:
3x null returns

Reply:
I've never personally seen it been an issue like. 

It would require loading the dash before any printers are added which is impossible with the current client setup. I'd personally say that should be sorted in another PR if it's that big of an issue. The dash view runner does expect most of the data been there which isn't very robust. 

The original calcs are here: 
https://github.com/OctoFarm/OctoFarm/blob/1d42e7066d76702a3aaf0d84e94d59988521461d/server_src/lib/dataFunctions/printerClean.js#L885-L894